packages/frontend/src/analytics.ts

```diff
@@ -6,20 +6,26 @@ import { config } from './config';
 let applicationInsights: ApplicationInsights | null = null;
 
 if (config.applicationInsightsInstrumentationKey) {
-  const reactPlugin = new ReactPlugin();
-  try {
-    applicationInsights = new ApplicationInsights({
-      config: {
-        instrumentationKey: config.applicationInsightsInstrumentationKey,
-        extensions: [reactPlugin as ITelemetryPlugin],
-        enableAutoRouteTracking: true,
-      },
-    });
-
-    applicationInsights.loadAppInsights();
-  } catch (error) {
-    console.error('Failed to initialize Application Insights:', error);
-    applicationInsights = null;
+  if (!import.meta.env.PROD) {
+    console.warn(
+      'ApplicationInsightsInstrumentationKey is defined but not enabled in non-production builds. Tracking is disabled.',
```

Comment:
Perhaps this warning doesn't make sense since it's only defined because it's not replaced? :)

Reply:
Flip? ```

if (import.meta.env.PROD) {
  if (config.applicationInsightsInstrumentationKey) {
    const reactPlugin = new ReactPlugin();
    try {
      applicationInsights = new ApplicationInsights({
        config: {
          instrumentationKey: config.applicationInsightsInstrumentationKey,
          extensions: [reactPlugin as ITelemetryPlugin],
          enableAutoRouteTracking: true,
        },
      });
      applicationInsights.loadAppInsights();
      console.log('Application Insights initialized successfully');
    } catch (error) {
      console.error('Failed to initialize Application Insights:', error);
      applicationInsights = null;
    }
  } else {
       console.warn('ApplicationInsightsInstrumentationKey is undefined. Tracking is disabled.');
}
}

```